numpy/ma/core.pyi

```diff
@@ -452,7 +453,13 @@ class MaskedArray(ndarray[_ShapeT_co, _DTypeT_co]):
     @property  # type: ignore[misc]
     def real(self): ...
     get_real: Any
-    def count(self, axis=..., keepdims=...): ...
+
+    # keep in sync with `np.ma.count`
+    @overload
+    def count(self, axis: None = None, keepdims: Literal[False] | _NoValueType = ...) -> int: ...  # type: ignore[overload-overlap]
```

Comment:
The `overload-overlap` error is a true positive, and an evitable one at that. The issue is that `int` and `NDArray[int_]` are incompatible (return) types, even though it could happen that both overloads are valid. So e.g. with `count(axis=None)` the return type is ambiguous.

One solution would be to make the parameter types non-overlapping, e.g. by requiring `keepdims: Literal[True]` in the second overload, or by removing `| None` from `axis` in the 2nd overload.